Link to video for 2024 CppNorth talk (#4578)
carbon-language/carbon-lang: README.md

Carbon focused talks from the community:

### 2024

-   Generic implementation strategies in Carbon and Clang, LLVM Developers'
    Meeting

>     ([announcement](https://discourse.llvm.org/t/announcing-the-2024-llvm-developers-meeting-program/81108))

Reviewer: I think we have slides for this now:

```suggestion
    ([slides](https://chandlerc.blog/slides/2024-llvm-generic-implementation/#/))
```

-   The Carbon Language: Road to 0.1, NDC {TechTown}
    ([slides](https://chandlerc.blog/slides/2024-ndc-techtown-carbon-road-to-0-dot-1))
-   How designing Carbon with C++ interop taught me about C++ variadics and
    overloads, CppNorth ([video](https://youtu.be/8SGMy9ENGz8),
    [slides](https://chandlerc.blog/slides/2024-cppnorth-design-stories))
-   Generic Arity: Definition-Checked Variadics in Carbon, C++Now

>     ([slides](https://youtu.be/Y_px536l_80))

Reviewer: ```suggestion
    ([video](https://youtu.be/Y_px536l_80))
```

-   [The Carbon Language: Road to 0.1](https://chandlerc.blog/slides/2024-ndc-techtown-carbon-road-to-0-dot-1),
    NDC {TechTown}

> -   [How designing Carbon with C++ interop taught me about C++ variadics and overloads](https://www.youtube.com/watch?v=8SGMy9ENGz8),

Reviewer 2: ```suggestion
-   [How designing Carbon with C++ interop taught me about C++ variadics and overloads](https://youtu.be/8SGMy9ENGz8),
```

Note, this is the link you get if you click "share" on the video, and most of the youtube links are using this format.

    CppNorth

>     ([slides](https://chandlerc.blog/slides/2024-cppnorth-design-stories))

Reviewer 2: To note as I did on #4576, there was a decision on #3158 not to keep slides for each talk. If you're going to change the format, please add slides for all talks below as well. However, I will suggest adopting the format suggested on #3158 if that's the path taken. (`Talk name, con ([video](youtube link), [slides](slides link))`)

You will probably need to research through historical talks to find slides. (many of these will be in the history, but you may also need to talk with authors to find the slides)

Reviewer: FWIW, I think it would be good to add slide links for other talks where we have both a video and slides, but it seems reasonable to me to do that in follow-up PRs.

However, I think the bigger thing is that it'd be good to be consistent about which links are videos and which are slides. Mostly this is because I think it could surprise folks clicking and expecting slides but to start getting audio from a video playing. (Maybe I'm the only one who routinely startles themselves with audio from a website accidentally?)

I could see two formats working here @josh11b:

```
- [Title](video-link) ([slides](slides-link))
- Title (no video yet, [slides](slides-link))
```

Or:

```
- Title ([video](video-link), [slides](slides-link))
- Title (no video yet, [slides](slides-link))
```

No strong preference between these. Happy if others have a preference to pick that one?

> -   [Generic Arity: Definition-Checked Variadics in Carbon](https://www.youtube.com/watch?v=Y_px536l_80),

Reviewer 2: ```suggestion
-   [Generic Arity: Definition-Checked Variadics in Carbon](https://youtu.be/Y_px536l_80),
```
